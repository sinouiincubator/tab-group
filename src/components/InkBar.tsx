import styled, { css } from 'styled-components';

const topCss = css`
  top: 0;
`;

interface Props {
  top?: boolean;
}

/**
 * 选择指示器
 */
const InkBar = styled.div.attrs({
  className: 'sinoui-ink-bar',
})<Props>`
  position: absolute;
  bottom: 0;
  ${(props) => props.top && topCss};
  left: 0;
  transition: ${(props) =>
    props.theme.transitions.create(['transform', 'width'])};
  height: 2px;
  background-color: ${(props) => props.theme.palette.primary[500]};
`;

export default InkBar;
