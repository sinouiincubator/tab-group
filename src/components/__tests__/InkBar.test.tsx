import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { defaultTheme } from '@sinoui/theme';
import 'jest-styled-components';
import InkBar from '../InkBar';

it('正确渲染InkBar', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={defaultTheme}>
        <InkBar />
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
