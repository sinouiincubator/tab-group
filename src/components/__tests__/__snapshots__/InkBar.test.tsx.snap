// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`正确渲染InkBar 1`] = `
.c0 {
  position: absolute;
  bottom: 0;
  left: 0;
  -webkit-transition: -webkit-transform 300ms cubic-bezier(0.4,0.0,0.2,1) 0ms,width 300ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  -webkit-transition: transform 300ms cubic-bezier(0.4,0.0,0.2,1) 0ms,width 300ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  transition: transform 300ms cubic-bezier(0.4,0.0,0.2,1) 0ms,width 300ms cubic-bezier(0.4,0.0,0.2,1) 0ms;
  height: 2px;
  background-color: #3f51b5;
}

<div
  className="c0 sinoui-ink-bar"
/>
`;
